src/components/MapView/Legends/index.tsx

```diff
@@ -193,6 +193,13 @@ function LegendItem({
   const map = useSelector(mapSelector);
   const analysisResult = useSelector(analysisResultSelector);
   const dispatch = useDispatch();
+  useEffect(() => {
+    // should this be here? Or somewhere more related to analysis?
```

Comment:
Confused why this is here. Also confused why we left what looks like an expensive operation without a useEffect.

Reply:
Did you find out? Or should I have a look @Ry-DS ?